pkgs/development/compilers/ghcjs-ng/8.6/dep-overrides.nix

```diff
@@ -1,11 +1,14 @@
-{ haskellLib }:
+{ lib, haskellLib, fetchFromGitHub }:
 
-let inherit (haskellLib) doJailbreak dontHaddock;
+let inherit (haskellLib) doJailbreak dontHaddock dontCheck;
 in self: super: {
+  ghcjs = super.ghcjs.override {
+    shelly = super.shelly_1_8_1;
+  };
```

Comment:
GHC 8.6.5 is used as the "boot" GHC for GHCJS. At some point, the `shelly` dependency was upgraded to 1.9.0, which introduced a breaking change in the API. `ghcjs` works fine with `shelly` 1.8.1.